### PR TITLE
Google Maps: Add Optional Consent Check before loading Google Maps API

### PR DESCRIPTION
**Pull request by AlexGStapleton**
Resolve https://github.com/siteorigin/so-widgets-bundle/issues/875

### CODE REVIEW

File: widgets/google-map/google-map.php [hunk 1]
```diff
@@ -466,6 +466,28 @@ function get_settings_form() {
...
+			'map_consent' => array(
+				'type' => 'checkbox',
+				'label'       => __( 'Require consent before loading Maps API', 'so-widgets-bundle' ),
+				'description' => __( 'Consent is required for the Google Maps widget to comply with regulations like DSGVO, or GDPR.', 'so-widgets-bundle' ),
+				'default' => false,
+			),
+
+			'map_consent_btn_text' => array(
+				'type' => 'text',
+				'label'       => __( 'Consent button text', 'so-widgets-bundle' ),
```
Comment (Misplon) [inline widgets/google-map/google-map.php:478]: An indentation fix is required here, thanks.

File: widgets/google-map/google-map.php [hunk 2]
```diff
@@ -466,6 +466,28 @@ function get_settings_form() {
...
+				'default' => __( 'I Consent', 'so-widgets-bundle' ),
+			),
+
+			'map_consent_notice' => array(
+				'type' => 'tinymce',
+				'label'       => __( 'Consent prompt text', 'so-widgets-bundle' ),
```
Comment (Misplon) [inline widgets/google-map/google-map.php:484]: Indentation check here too.

File: widgets/google-map/tpl/js-map.php [hunk 3]
```diff
@@ -6,9 +6,16 @@
  * @var $fallback_image_data
  */
 ?>
+<?php if ( $map_consent ): ?>
```
Comment (Misplon) [inline widgets/google-map/tpl/js-map.php:9]: Ideally need a space before `:`.

**Comment by Misplon** [comment]
- Clicking anywhere in the content message reveals the map.
- Can we perhaps have the map in the background with an opacity overlay?
- Could we perhaps simplify the message? I like Borlab's option:

_By loading the map, you agree to OpenStreetMap Foundation's privacy policy.
Learn more (linked)
Load map_

Borlabs_Cookie_Iframe_Demo_–_WordPress_Cookie_Plugin

**Comment by AlexGStapleton** [comment]
> Can we perhaps have the map in the background with an opacity overlay?

Not the JavaScript version, no. We could show the static map, but that won't work for most people without them adjusting their API key, and it would almost double API usage.

We could however show a background image with a generic map image - which appears to be the case in your image. If you would like to go that route, we'll need to have a generic placeholder image created. We should also add a setting that allows users to change the background image.

> Could we perhaps simplify the message? I like Borlab's option:

Yeah, we can do that. Do you want this to be editable, or do you want it to "Learn More"?
How about we only show the Read More link if the site has a privacy policy set up? Is that text editable?

**Comment by Misplon** [comment]
> If you would like to go that route, we'll need to have a generic placeholder image created. We should also add a setting that allows users to change the background image.

Super, let's do that, thanks.

> Yeah, we can do that. Do you want this to be editable, or do you want it to "Learn More"?
How about we only show the Read More link if the site has a privacy policy set up? Is that text editable?

I'd be keen to use a learn more link but make it all editable so users can change the text and destination.

**Comment by AlexGStapleton** [comment]
I sent you a few questions over slack regarding settings. Based on your answer I'll make some further adjustments to the design - specifically, change the color of the prompt text.

Do we have an image we can use for the consent prompt background?

**Comment by Misplon** [comment]
> Do we have an image we can use for the consent prompt background?
I'll try to find some sort of generic map illustration that we can use.

**Comment by Misplon** [comment]
We might need to get a vector illustrated for us. Perhaps we can use something like the below for now?

snazzy-image-3x

**Comment by Misplon** [comment]
Thanks, mate. Please, can you check up on:

`Parse error: syntax error, unexpected ',', expecting ';' in /wp-content/plugins/so-widgets-bundle/widgets/google-map/google-map.php on line 490`

**Comment by AlexGStapleton** [comment]
Sorry about thar. I excluded a setting from one of the last commits and let that slip through.

**Comment by Misplon** [comment]
Thanks, mate. I've made some minor styling adjustments. The Less doesn't seem to be writing for me locally. How does that look for you?

**Review by Misplon — APPROVED** [review]

**Review by Misplon — APPROVED** [review]

**Review by gregpriday — APPROVED** [review]

**Pull request merged**